### PR TITLE
[#10105] Add set max to Send transaction

### LINKED ISSUE #10105: Add set max to Send transaction
# Feature Issue
### User Story
As a user I want to be able to quickly move all my funds, because, for example I lost my seed phrase

### Description
Designs on [Figma](https://www.figma.com/file/XUehMnhyD1FGcWzvGz6SXqvh/Wallet?node-id=5849%3A42299)

* [ ] Adds available amount in wallet for selected asset
* [ ] In case ETH is selected, the user will need to manually edit the total to reserve some funds for gas. This design does not resolve this issue yet
* [ ] When user taps on `Send all`
* [ ] Amount is updated to available assets in account

1!UNITO-UNDERSCORE!1 Chat

### Acceptance Criteria
Implementation follows design

### Notes
Best implemented after or paired with https://github.com/status-im/status-react/issues/10103

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes #10105

Also added fiat value

### CODE REVIEW

File: src/status_im/ui/screens/wallet/send/views.cljs [hunk 1]
```diff
@@ -70,40 +67,56 @@
...
+                                                  {:content        sheets/choose-recipient
+                                                   :content-height 200}]))
+      :accessories         [:chevron]}]))
+
+(defn set-max [token]
+  [react/touchable-highlight
+   {:on-press
+    #(when token
+       (re-frame/dispatch [:wallet.send/set-max-amount token]))}
+   [react/view {:height            35 :border-radius 40 :background-color colors/blue-light
```
Comment (rasom) [inline src/status_im/ui/screens/wallet/send/views.cljs]: This looks unreadable to me, do you mind to add comas here or keep one pair per line?

Comment (flexsurfer) [reply src/status_im/ui/screens/wallet/send/views.cljs]: done

**Comment by Serhy** [comment]
Tested with iOS13, Android8, Android 9 3rd build (`62871f0`)
1) On iOS there is a comma decimal separator. When asset is not integer, - fiat value is `NaN`. It's NaN also when amount is invalid input (like `1.234.56`). Not sure what is better to display as fiat value when invalid input entered. @errorists? 
2) In the 1-1 chat -> Send TX there is no `Set max` option (but fiat value is present ok)
For above two screenshot:

Screenshot 2020-06-26 at 14 37 53


3) I guess we may add fiat value in `Request TX` option in 1-1 chat too. No fiat value at the moment there.

**Comment by flexsurfer** [comment]
thanks @Serhy 
2. it's intentional because when we set max we decrease on fee, so it doesn't make sense if you send in chat because you need to request address, and later fee will be different
3. yes we can

**Comment by flexsurfer** [comment]
actually this set max is tricky for eth, if you set max and then change fee, it won't be max anymore

**Comment by errorists** [comment]
How do I get the NaN again? I couldn't seem to replicate it, anyway that shouldn't be displayed under any circumstances, should be a 0.0
wrt to max and network fees, imo it's okay, it's just if after the change in fee, the total value is greater than available amount we need to crap out with an error and disable proceeding to next step.

**Comment by Serhy** [comment]
> actually this set max is tricky for eth, if you set max and then change fee, it won't be max anymore

The way Trust Wallet works is: Set Max = sets all available ETH -> Next -> and on the next screen it becomes (Max - Fee). Further changes of fee deducts/increases this `Max` value thus making an ability to send everything.

If I'm correct we have a fixate ETH amount on the `Sending... ...... ETH` screen after specifying Asset-From-To on previous screen.

**Comment by flexsurfer** [comment]
thanks @Serhy  2 and 3 fixed

**Comment by oskarth** [comment]
Small request - could you please rebase to latest master so this commit https://github.com/status-im/status-react/commit/b5fda1234e5447cbc626059dfc78fcfbabc9e926 gets included? This means we aren't running E2E tests on eth.prod cluster anymore, and since this happens on every push, the impact is quite big. This ensures we have more accurate metrics going forward from Jul 1 onward, which would be awesome. See https://discuss.status.im/t/user-growth-and-retention/1782 for more

**Comment by Serhy** [comment]
Tested on iOS13, Android8.1, Android 9 - all above issues fixed, can set max value for ETH/token.

@flexsurfer I just noticed the currency in newly introduced fiat value always set to `$` disregard the fiat currency what is set when sending transaction. 
Screenshot 2020-07-01 at 17 41 42

**Comment by flexsurfer** [comment]
@Serhy fixed, hided it for all cases

**Comment by Serhy** [comment]
Looks good now! Meaningful currency code everywhere.
E2e test will be fixed separately (not this PR issue)

**Pull request merged**